gensim/models/word2vec.py

```diff
@@ -1112,7 +1112,7 @@ def update_weights(self):
         # randomize the remaining words
         for i in xrange(len(self.wv.syn0), len(self.wv.vocab)):
             # construct deterministic seed from word AND seed argument
-            newsyn0[i-len(self.wv.syn0)] = self.seeded_vector(self.wv.index2word[i] + str(self.seed))
+            newsyn0[i-len(self.wv.syn0)] = self.seeded_vector(utils.to_unicode(self.wv.index2word[i]) + str(self.seed))
```

Comment:
Is this change inside `word2vec.py`, and similar below, absolutely necessary? It seems if the caller has provided unicode strings in their corpus, the contents of `index2word()` will already be unicode. But it seems unless `seeded_vector()` is otherwise throwing an exception, it really doesn't matter whether this string-used-for-deterministic-randomization is unicode or not.